### PR TITLE
bcoin: v2.0.0 (new formula)

### PR DESCRIPTION
**Pull request by tynes**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----
This pull request adds [bcoin](https://github.com/bcoin-org/bcoin), a JavaScript based Bitcoin full node. A formula that was recently merged in ([hsd](https://github.com/handshake-org/hsd)) is a fork of this codebase.
See: https://github.com/Homebrew/homebrew-core/pull/51014

### CODE REVIEW

File: Formula/bcoin.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
+require "language/node"
+
+class Bcoin < Formula
+  desc "Javascript bitcoin library for node.js and browsers"
+  homepage "https://bcoin.io"
+  url "https://github.com/bcoin-org/bcoin/archive/v2.0.0.tar.gz"
+  sha256 "ff99b735d3b23624455d72eada0de7f7ba4207f4a4cc0cab963ca2036de254a3"
+  depends_on "python" => :build
+  depends_on "node@10"
```
Comment (SMillerDev) [inline Formula/bcoin.rb:9]: Please try and get it to build with the latest node.js. Using old versions in new formulae should be avoided.

File: Formula/bcoin.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
+require "language/node"
+
+class Bcoin < Formula
+  desc "Javascript bitcoin library for node.js and browsers"
+  homepage "https://bcoin.io"
+  url "https://github.com/bcoin-org/bcoin/archive/v2.0.0.tar.gz"
+  sha256 "ff99b735d3b23624455d72eada0de7f7ba4207f4a4cc0cab963ca2036de254a3"
+  depends_on "python" => :build
```
Comment (SMillerDev) [inline Formula/bcoin.rb:8]: Please try with
```suggestion
  depends_on "python@3.8" => :build
```

Comment (tynes) [reply Formula/bcoin.rb]: Updated to use `python@3.8`, I originally thought that `node-gyp` required python 2.7 but looks like it works fine with 3.8 according to the `node-gyp` docs.

File: Formula/bcoin.rb (hunk 1 above)
Comment (tynes) [reply Formula/bcoin.rb]: Updated to use `node`

**Comment by SMillerDev** [comment]
Can you squash your commits?

**Comment by tynes** [comment]
@SMillerDev commits are squashed, thank you for your time.

**Comment by tynes** [comment]
The error from the logs is here:
```
==> brew audit bcoin --online --new-formula
Error: GitHub Bad credentials:HOMEBREW_GITHUB_API_TOKEN may be invalid or expired; check:
  https://github.com/settings/tokens
==> FAILED
```
I think I remember all of the tests passing before I force pushed the squashed commits.

**Comment by SMillerDev** [comment]
Thanks @tynes! Without contributions like yours it'd be impossible to keep homebrew going with the high standards that users have come to expect from the project. You can feel good knowing that you've made the world a tiny bit better for homebrew users around the world! 👍 🎉

**Pull request closed without merging**